### PR TITLE
Filter components

### PR DESCRIPTION
**Pull request by L0wry**
the component list now has the ability to give components nested structure
This is done by using a `/` in the component name or story

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "lint:e2eTests": "eslint e2eTests /--fix",
     "prettier:diff": "prettier --list-different '**/*.js'",
     "prettier:write": "prettier --write '**/*.js'",
-    "prepare": "babel packages/test-helpers/src --out-dir packages/test-helpers/lib"
+    "prepare": "babel packages/test-helpers/src --out-dir packages/test-helpers/lib",
+    "copy": "APPDIR=e2eTests/node_modules/@times-components/fructose/packages && rm -rf $APPDIR && cp -r ./packages $APPDIR"
```
Comment (rjanjua) [inline package.json]: nice

File: packages/app/src/components/navigation/menuListComponent.js [hunk 2]
```diff
@@ -1,72 +1,75 @@
...
-import {
-  View,
-  FlatList,
-  Text,
-  TouchableHighlight,
-  StyleSheet
-} from "react-native";
+import { View, Text, StyleSheet } from "react-native";
 import PropTypes from "prop-types";
+import NestedListview, { NestedRow } from "react-native-nested-listview";
```
Comment (rjanjua) [inline packages/app/src/components/navigation/menuListComponent.js]: NestedListView

File: packages/app/src/components/navigation/menuListComponent.js [hunk 3]
```diff
@@ -1,72 +1,75 @@
...
 
-const menuHeader = headerText => (
-  <Text style={styles.menuHeader}>{headerText}</Text>
-);
-
-const menuSeparator = () => <View style={styles.menuSeparator} />;
+const MenuSeparator = () => <View style={styles.menuSeparator} />;
 
-const prepareMenuItems = menuArray => menuArray.map(item => ({ key: item }));
+const menuNode = (node, level) => {
```
Comment (rjanjua) [inline packages/app/src/components/navigation/menuListComponent.js]: MenuNode

Comment (rjanjua) [reply packages/app/src/components/navigation/menuListComponent.js]: use props

File: packages/app/src/components/navigation/menuDataGenerator.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    if (a.title < b.title) {
+      return -1;
+    }
+    if (a.title > b.title) {
+      return 1;
+    }
+    return 0;
+  });
+
+  return sortedArray.reduce((collection, currentObject, index) => {
```
Comment (shendriksen) [inline packages/app/src/components/navigation/menuDataGenerator.js]: Should this be a 'map' more than a reduce as items are not being reduced?

**Pull request merged**